python/triton/runtime/autotuner.py

```diff
@@ -38,17 +38,36 @@ def __init__(self, fn, arg_names, configs, key, reset_to_zero, prune_configs_by:
             self.configs = configs
         self.key_idx = [arg_names.index(k) for k in key]
         self.cache = {}
-        # hook to reset all required tensor to zeros before relaunching a kernel
-        self.hook = lambda args: 0
+        self.arg_names = arg_names
+        
+        # Reset to zero or restore values
+        self.reset_idx = []
         if reset_to_zero is not None:
             self.reset_idx = [arg_names.index(k) for k in reset_to_zero]
-
-            def _hook(args):
+        self.restore_idx = []
+        self.restore_copies = []
+        if restore_value is not None:
+            self.restore_idx = [arg_names.index(k) for k in restore_value]
+        
+        # Hook to reset or restore for required tensors
+        self.pre_hook = lambda args, reset_only=False: 0
+        self.post_hook = lambda args: 0
+        if len(self.reset_idx) > 0 or len(self.restore_idx) > 0:
+            def _pre_hook(args, reset_only=False):
                 for i in self.reset_idx:
                     args[i].zero_()
-            self.hook = _hook
-        self.arg_names = arg_names
-        # prune configs
+                if not reset_only:
+                    self.restore_copies = []
```

Comment:
This line can be removed?

Reply:
This line is for saving GPU memory, if we directly write `self.restore_copies = [args[i].clone() for i in self.restore_idx]`, then actually there is a time that Python interpreter has both old `self.restore_copies` and new values. But anyway, removing it is not a big deal :)

Reply:
I see, that's interesting